src/Components/Patient/PatientRegister.tsx

```diff
@@ -523,7 +528,29 @@ export const PatientRegister = (props: PatientRegisterProps) => {
         case "name":
         case "gender":
         case "date_of_birth":
-          errors[field] = RequiredFieldValidator()(form[field]);
+          if (!form["age"] && !form["date_of_birth"]) {
+            errors[field] = "Please fill the date of birth or age";
+            errors[field] = "Please fill the date of birth or age";
+          } else if (!form["age"]) {
+            errors[field] = RequiredFieldValidator()(form[field]);
+            form["year_of_birth"] = form[field]
+              ? new Date(form[field]).getFullYear()
+              : null;
+          }
+          return;
+        case "age":
+          if (form[field]) {
+            if (form[field] >= 0) {
+              form["year_of_birth"] = new Date().getFullYear() - form[field];
+              form["date_of_birth"] = new Date(
+                new Date().getFullYear() - form[field],
+                0,
+                1
+              );
```

Comment:
We need to have a clear distinction on what was captured from the user.

If we coalesce `date_of_birth` with January 1st of input year from front-end, we'll never know if the date_of_birth was actually January 1st of that year or just that year.

Set `date_of_birth` to null if form is taking age as input.